File: DSA/Gautam_Shyamnani_590013078/Day11Question1.cpp
#include <iostream>
#include <string>
#include <cctype>
using namespace std;

bool isPalindrome(string s) {
    int st = 0, end = s.length() - 1;

    while (st < end) {
        
        while (st < end && !isalnum(s[st])) ++st;
        while (st < end && !isalnum(s[end])) --end;

        
        if (tolower(s[st]) != tolower(s[end])) {
            return false;
        }

        ++st;
        --end;
    }

    return true;
}

int main() {
    string s;
    cout << "Enter a string: ";
    getline(cin, s); 
    if (isPalindrome(s)) {
        cout << "Is a palindrome" << endl;
    } else {
        cout << "Not a palindrome" << endl;
    }

    return 0;
}

File: DSA/Stuti_Jain_590012149/Day_11_Question_1.py
string=input("Enter the string: ")
case=""
for char in string:
    if ('a'<=char <= 'z') or ('A' <= char <= 'Z') or ('0'<= char <= '9'):
        case= case+ char


if(case==case[::-1]):
    print("Palindrome")
else:
    print("Not a Palindrome")
